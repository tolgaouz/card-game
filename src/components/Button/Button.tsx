import clsx from 'clsx';
import React from 'react';

interface ButtonProps extends React.HTMLProps<HTMLButtonElement> {
  variant?: 'primary' | 'outline';
  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
  type?: 'button' | 'submit' | 'reset';
}

const Button: React.FC<React.PropsWithChildren<ButtonProps>> = ({
  variant = 'primary',
  onClick,
  children,
  className,
  ...restProps
}) => {
  const getVariantClasses = {
    primary: 'bg-yellow-300',
    outline: 'border-2 text-yellow-300 border-yellow-300',
  };
  return (
    <button
      className={clsx(getVariantClasses[variant], 'font-alfa rounded-xl', className)}
      type="button"
      onClick={onClick}
      {...restProps}
    >
      {children}
    </button>
  );
};

export default Button;
