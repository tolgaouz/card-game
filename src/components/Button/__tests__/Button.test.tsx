import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from '../Button';

describe('Button', () => {
  it('renders correctly', () => {
    const { container } = render(<Button>Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it('calls onClick function successfully', () => {
    const onClick = jest.fn();
    const { getByText } = render(<Button onClick={onClick}>Button</Button>);
    userEvent.click(getByText('Button'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('applies correct classes when variant is outlined', () => {
    const { getByText } = render(<Button variant="outline">Button</Button>);
    const button = getByText('Button');
    expect(button).toHaveClass('border-2 text-yellow-300 border-yellow-300');
  });

  it('applies correct classes when variant is primary', () => {
    const { getByText } = render(<Button variant="primary">Button</Button>);
    const button = getByText('Button');
    expect(button).toHaveClass('bg-yellow-300');
  });

  it('passes down props to button element', () => {
    const { getByText } = render(
      <Button variant="primary" data="test">
        Button
      </Button>
    );
    const button = getByText('Button');
    expect(button.getAttribute('data')).toBe('test');
  });
});
