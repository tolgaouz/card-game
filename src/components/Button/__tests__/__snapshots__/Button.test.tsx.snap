// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div>
  <button
    class="bg-yellow-300 font-alfa rounded-xl"
    type="button"
  >
    Button
  </button>
</div>
`;
